File: components/bad-factors/worker.ts
type IWorkerInput = BigInt;

export type CustomWorker = Omit<Worker, 'postMessage'> & {
	postMessage(command: IWorkerInput): void;
};

type IWorkerResult = number[];

export interface IWorkerReturn {
	data: IWorkerResult;
}

onmessage = async e => {
	const targetNumber = Number(e.data as IWorkerInput);

	const midPoint = Math.floor(targetNumber / 2);

	const results: IWorkerResult = [];
	for (let i = 1; i <= midPoint; i++) {
		const result = targetNumber / i;
		if (result % 1 === 0) results.push(i);
	}

	return postMessage(results);
};

File: components/bad-factors/worker.go
package main

import "math"

func main() {
	println("Initialised Go")
}

func getFactors(input int) []int {
	midPoint := int(math.Floor(float64(input / 2)))

	var results []int = []int{}

	i := 1
	for ; i <= midPoint; i++ {
		result := input / i
		if result%1 == 0 {
			results = append(results, i)
		}
	}

	return results
}
